okhttp/src/main/java/okhttp3/internal/Util.java

```diff
@@ -498,6 +499,14 @@ public static int checkDuration(String name, long duration, TimeUnit unit) {
     return (int) millis;
   }
 
+  public static int checkDuration(String name, Duration duration) {
```

Comment:
I don't think we need this method. Have the duration overloads call `sameMethod(duration.toMillis(), TimeUnit.MILLISECONDS)` so that they're just JDK 8 sugar on the JDK 6 methods.

Reply:
Do you mean remove this method
``` java
  public static int checkDuration(String name, Duration duration) {
    if (duration == null) throw new NullPointerException("duration == null.");
    long millis = duration.toMillis();
    if (millis < 0) throw new IllegalArgumentException(name + " < 0");
    if (millis > Integer.MAX_VALUE) throw new IllegalArgumentException(name + " too large.");
    return (int) millis;
  }
```
and call like this
``` java
    @IgnoreJRERequirement
    public Builder writeTimeout(Duration duration) {
//      writeTimeout = checkDuration("timeout", duration);
      writeTimeout = checkDuration("timemeout", duration.toMillis(), TimeUnit.MILLISECONDS);
      return this;
    }
```

Reply:
Precisely